contracts/router/modules/pool/traderjoe/TraderJoeModule.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import {IERC20} from "@openzeppelin/contracts-4.8.0/token/ERC20/IERC20.sol";
+
+import {IndexedToken, IPoolModule} from "../../../interfaces/IPoolModule.sol";
+import {ILBRouter} from "../../../interfaces/traderjoe/ILBRouter.sol";
+import {UniversalTokenLib} from "../../../libs/UniversalToken.sol";
+
+import {OnlyDelegateCall} from "../OnlyDelegateCall.sol";
+
+/// @notice PoolModule for Trader Joe LBPairs
+/// @dev Implements IPoolModule interface to be used with pools added to LinkedPool router
+abstract contract TraderJoeModule is OnlyDelegateCall, IPoolModule {
+    using UniversalTokenLib for address;
+
+    ILBRouter public immutable lbRouter;
+
+    constructor(address _lbRouter) {
+        lbRouter = ILBRouter(_lbRouter);
+    }
+
+    /// @notice Trader Joe LBPair version this module accomodates
+    function version() public view virtual returns (ILBRouter.Version);
+
+    /// @notice Bin step for the given pool and version this module accomodates
+    function _binStep(address pool) internal view virtual returns (uint256);
+
+    /// @inheritdoc IPoolModule
+    function poolSwap(
+        address pool,
+        IndexedToken memory tokenFrom,
+        IndexedToken memory tokenTo,
+        uint256 amountIn
+    ) external returns (uint256 amountOut) {
+        assertDelegateCall();
+        address tokenIn = tokenFrom.token;
+        tokenIn.universalApproveInfinity(address(lbRouter), amountIn);
+
+        // @dev https://docs.traderjoexyz.com/guides/swap-tokens#1-swap-10-usdc-for-usdt-using-swapexacttokensfortokens-with-no-intermediate-swap-paths
```

Comment:
Let's cut the link a bit here to avoid linter shouting out for having "line too long"
```suggestion
        /// @dev https://docs.traderjoexyz.com/guides/swap-tokens#code-examples
```